core/src/main/scala/org/ensime/indexer/DatabaseService.scala

```diff
@@ -28,83 +32,76 @@ class DatabaseService(dir: File) extends SLF4JLogging {
     (ds, Database.forDataSource(ds))
   }
 
-  def shutdown(): Unit = {
+  def shutdown()(implicit ec: ExecutionContext): Future[Unit] = for {
     // call directly - using slick withSession barfs as it runs a how many rows were updated
     // after shutdown is executed.
-    db.createSession().createStatement() execute "shutdown;"
-    datasource.close()
-  }
+    _ <- db.run(sqlu"shutdown")
+    _ <- db.shutdown
+    _ = datasource.close()
+  } yield ()
 
   if (!dir.exists) {
     log.info("creating the search database")
     dir.mkdirs()
-    db withSession { implicit s =>
-      (fileChecks.ddl ++ fqnSymbols.ddl).create
-    }
+    Await.result(
+      db.run(
+        (fileChecks.schema ++ fqnSymbols.schema).create
+      ),
+      DB_CREATE_TIMEOUT
```

Comment:
yeah. The end effect to the users is the same because they'll still be waiting forever in their ENSIME client even if this throws.